src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -92,17 +105,17 @@ static public StackValue CreateFromType(TypeDesc type)
                 case TypeFlags.UInt16:
                 case TypeFlags.Int32:
                 case TypeFlags.UInt32:
-                    return CreatePrimitive(StackValueKind.Int32);
+                    return CreatePrimitive(StackValueKind.Int32, type);
```

Comment:
IIRC, all previous verifiers have been tracking the verification types only on the IL stack. Also, the ECMA spec suggest it by saying things like `"The “VerificationType” is a corresponding type used for type compatibility rules in verification.`